src/kaocha/plugin/notifier.clj

```diff
@@ -6,14 +6,22 @@
             [kaocha.result :as result]
             [kaocha.shellwords :refer [shellwords]]
             [clojure.string :as str]
-            [clojure.java.io :as io])
-  (:import [java.nio.file Files]))
+            [clojure.java.io :as io]
+            [slingshot.slingshot :refer [throw+]])
+  (:import [java.nio.file Files]
+           [java.io IOException] ))
 
 ;; special thanks for terminal-notify stuff to
 ;; https://github.com/glittershark/midje-notifier/blob/master/src/midje/notifier.clj
 
 (defn exists? [program]
-  (= 0 (:exit (sh "which" program))))
+  (let [cmd (if 
+              (re-find #"Windows" (System/getProperty "os.name"))
+              "where.exe" "which" )]
+    (try 
+      (= 0 (:exit (sh cmd program)))
+      (catch IOException e  ;in the unlikely event where.exe or which isn't available
+        (throw+ {:kaocha/early-exit 0} e)))))
```

Comment:
I'm not aware of any platforms that wouldn't be covered by this logic, so I could see an argument for this code is unnecessary. On the other hand, I think shelling out has an inherent risk, particularly in the context of something running on a VM, so maybe all calls to sh should have a similar guard?

Reply:
I agree, shelling out can always error for whatever reason. But maybe throwing a `:kaocha/early-exit` is a bit much here. We can still run tests without notifications, so maybe just outputting a warning is better.